docs/source/configuration/validation.md

```diff
@@ -0,0 +1,297 @@
+This extensibility is based on the Volto registry.
+It applies to content types, custom programatically generated forms, and blocks schema settings.
+The mechanism serializes all of them according to the [JSON schema standard](https://json-schema.org/draft/2020-12/json-schema-validation).
+Finally Volto generates the form from the serialization.
+
+
+## Register a validator
+
+You can register a validator using the `registerUtility` method in the registry API from your add-on configuration.
+
+
+### Register and declare a simple validator
+
+This section describes how to validate a field with a specific validator, a common use case.
+Volto also provides some default validators.
+
+```{seealso}
+{ref}`voltos-default-validators-label`
+```
+
+#### Volto custom forms and block schema forms
+
+When you define custom forms and block schema forms programatically, you can register a custom validator using the `format` property in your core using JSON schema.
+
+The following example shows how to create the schema for a block.
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom URL field',
+      description: '',
+      format: 'url'
+    },
+  },
+  required: [],
+};
+```
+
+You should register the `url` named validator as a Volto validator utility.
+In the following example, the `urlValidator` method validator will be applied for the block field `customField` in the previous example.
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'url',
+  method: urlValidator,
+})
+```
+
+
+#### Content types
+
+You can also specify the `format` of content types using the schema hints in the backend using `frontendOptions`.
+
+```python
+from plone.supermodel import model
+from zope import schema
+
+class IMyContent(model.Schema):
+    directives.widget(
+        "customField",
+        frontendOptions={
+            "format": "url",
+        },
+    )
+    customField = schema.TextLine(
+        title="Custom URL field",
+        required=False,
+    )
+    # Rest of your content type definition
+```
+
+The response from `plone.restapi` will be something like the following.
+It is slightly different from blocks JSON schema, but the validation engine will behave the same.
+The `urlValidator` method validator will be applied for the content type field `customField` from the earlier example.
+
+```json
+{
+  "properties": {
+    "customField": {
+      "title": "Custom URL field",
+      "widgetOptions": {
+        "frontendOptions": {
+          "format": "url"
+        }
+      }
+    }
+  }
+}
+```
+
+
+### Advanced scenarios
+
+In case you need more granularity, and you don't have access to modify the existing implementation of the JSON schema definitions for existing content types, blocks, or forms, you can use the following advanced validator registrations, using `field`, `widget`, `behaviorName`, or `blockType` validator registrations.
```

Comment:
I don't think "more granularity" is the right way to describe this. Nothing is more granular (specific) than assigning one specific validator to a specific field using `format`. This is less granular: it's a generic approach to register validators that apply to multiple fields automatically based on their field/widget/behaviorName/blockType. Right?

Reply:
@davisagli I tried to improve it, take a look if now it's better.